src/app/features/ledger/hooks/use-when-reattempt-ledger-connection.ts

```diff
@@ -1,15 +1,19 @@
-import { useEffect } from 'react';
 import { useLocation } from 'react-router-dom';
 
+import { useOnMount } from '@app/common/hooks/use-on-mount';
+
 export const immediatelyAttemptLedgerConnection = 'immediatelyAttemptLedgerConnection' as const;
 
 export function useWhenReattemptingLedgerConnection(fn: () => void) {
   const location = useLocation();
 
-  useEffect(() => {
+  useOnMount(() => {
     const state: any = location.state;
     if (typeof state !== 'object' || state === null) return;
-    if (state[immediatelyAttemptLedgerConnection]) fn();
+    if (state[immediatelyAttemptLedgerConnection]) {
+      // hack to call function on mount
+      setTimeout(fn);
```

Comment:
is this where `useLayoutEffect` is sometimes used, instead of pushing to back of callstack?

Reply:
prob, but you cannot use `useLayoutEffect` in callback 🤔